src/control/server/config.go

```diff
@@ -380,27 +376,47 @@ func saveActiveConfig(log logging.Logger, config *Configuration) {
 }
 
 // Validate asserts that config meets minimum requirements.
-func (c *Configuration) Validate() (err error) {
+func (c *Configuration) Validate(log logging.Logger) (err error) {
 	// append the user-friendly message to any error
 	// TODO: use a fault/resolution
 	defer func() {
 		if err != nil {
 			examplesPath, _ := c.ext.getAbsInstallPath(relConfExamplesPath)
-			err = errors.WithMessage(err, msgBadConfig+examplesPath)
+			err = errors.WithMessage(FaultBadConfig,
+				err.Error()+", examples: "+examplesPath)
 		}
 	}()
 
 	if c.Fabric.Provider == "" {
-		return errors.New(msgConfigNoProvider)
+		return FaultConfigNoProvider
 	}
 
 	// only single access point valid for now
 	if len(c.AccessPoints) != 1 {
-		return errors.New(msgConfigBadAccessPoints)
+		return FaultConfigBadAccessPoints
+	}
+	for i := range c.AccessPoints {
+		// apply configured control port if not supplied
+		host, port, err := common.SplitPort(c.AccessPoints[i], c.ControlPort)
+		if err != nil {
+			return errors.Wrap(FaultConfigBadAccessPoints, err.Error())
+		}
+
+		// warn if access point port differs from config control port
+		if strconv.Itoa(c.ControlPort) != port {
+			log.Debugf("access point (%s) port (%s) differs from control port (%d)",
+				host, port, c.ControlPort)
+		}
```

Comment:
Shouldn't this be logged at Error at least? No one will see this message unless they're running with debug logging enabled, and that won't be the usual case in production. I guess we can't just return an error here because technically it's possible to run different control plane instances on different ports, although that's kind of wacky IMO.

Reply:
it's not an error, just something people should be notified of if they are looking